server/routes/users_all.js

```diff
@@ -1,10 +1,62 @@
 const router = require('express').Router();
+const { pickBy, identity, pipe, omit, assoc } = require('ramda');
+const moment = require('moment');
 const usersAll = require('../database/queries/users_all');
+const Joi = require('joi');
+const { validate } = require('../shared/middleware');
 
-router.get('/', (req, res, next) => {
-  usersAll(req.app.get('client:psql'), req.auth.cb_id)
-    .then(users => res.send({ result: users }))
-    .catch(err => next(err));
+const schema = {
+  query: {
+    offset: Joi.number()
+      .integer()
+      .min(0)
+      .required(),
+    sort: Joi.any().valid(
+      '',
+      'id',
+      'name',
+      'email',
+      'gender',
+      'yob',
+      'email_consent',
+      'sms_consent'
+    ),
+    sex: Joi.any().valid('', 'male', 'female', 'prefer not to say'),
+    age: Joi.any().valid('', '0-17', '18-35', '35-50', '51-69', '70+'),
+  },
+};
+
+const ageRange = ageString =>
+  ageString
+    .replace('+', '')
+    .split('-')
+    .map(x => moment().year() - x)
+    .reverse();
+
+const removeProperties = omit(['sort', 'offset', 'age']);
+const removeEmpty = pickBy(identity);
+
+router.get('/', validate(schema), async (req, res, next) => {
+  try {
+    const query = req.query;
+
+    const addCbId = assoc('cb_id', req.auth.cb_id);
+    const queryPipe = pipe(removeProperties, removeEmpty, addCbId);
```

Comment:
Is `queryPipe` a good name here? Doesn't tell me what it's for. What it's actually doing is constructing the `where` object, right? What about `createWhereOpts` or `createWhereObj` or `createWhereClauseOpts` or something like that.